From 99c9f6966e0690648d99641337ca10f9f3e1e4de Mon Sep 17 00:00:00 2001
From: Enrico Berti <hello@enricoberti.com>
Date: Thu, 12 Mar 2015 22:20:23 -0500
Subject: [PATCH 1067/1173] HUE-2625 [oozie] Workflow dashboard completed
 table is being refreshed for nothing

Fixed bug on data length
---
 .../templates/dashboard/list_oozie_bundles.mako    |    2 +-
 .../dashboard/list_oozie_coordinators.mako         |    2 +-
 .../templates/dashboard/list_oozie_workflows.mako  |    3 +--
 3 files changed, 3 insertions(+), 4 deletions(-)

diff --git a/apps/oozie/src/oozie/templates/dashboard/list_oozie_bundles.mako b/apps/oozie/src/oozie/templates/dashboard/list_oozie_bundles.mako
index 35e06ef..8eec4a9 100644
--- a/apps/oozie/src/oozie/templates/dashboard/list_oozie_bundles.mako
+++ b/apps/oozie/src/oozie/templates/dashboard/list_oozie_bundles.mako
@@ -373,7 +373,7 @@ ${layout.menubar(section='bundles', dashboard=True)}
         if (data.jobs.length != numRunning) {
           refreshCompleted();
         }
-        numRunning = data.length;
+        numRunning = data.jobs.length;
 
         window.setTimeout(refreshRunning, 20000);
       });
diff --git a/apps/oozie/src/oozie/templates/dashboard/list_oozie_coordinators.mako b/apps/oozie/src/oozie/templates/dashboard/list_oozie_coordinators.mako
index 0519926..a80ca7d 100644
--- a/apps/oozie/src/oozie/templates/dashboard/list_oozie_coordinators.mako
+++ b/apps/oozie/src/oozie/templates/dashboard/list_oozie_coordinators.mako
@@ -425,7 +425,7 @@ ${layout.menubar(section='coordinators', dashboard=True)}
         if (data.jobs.length != numRunning) {
           refreshCompleted();
         }
-        numRunning = data.length;
+        numRunning = data.jobs.length;
         window.setTimeout(refreshRunning, 20000);
       });
     }
diff --git a/apps/oozie/src/oozie/templates/dashboard/list_oozie_workflows.mako b/apps/oozie/src/oozie/templates/dashboard/list_oozie_workflows.mako
index 570ea5d..73632fb 100644
--- a/apps/oozie/src/oozie/templates/dashboard/list_oozie_workflows.mako
+++ b/apps/oozie/src/oozie/templates/dashboard/list_oozie_workflows.mako
@@ -399,11 +399,10 @@ ${ layout.menubar(section='workflows', dashboard=True) }
         if (data.jobs.length == 0) {
           runningTable.fnClearTable();
         }
-
         if (data.jobs.length != numRunning) {
           refreshCompleted();
         }
-        numRunning = data.length;
+        numRunning = data.jobs.length;
 
         window.setTimeout(refreshRunning, 5000);
       });
-- 
1.7.9.5

